include/arc_utilities/dijkstras.hpp

```diff
@@ -670,6 +670,155 @@ namespace arc_dijkstras
             }
     };
 
+    template<typename NodeValueType, typename Allocator=std::allocator<NodeValueType>>
+    class SimpleGraphAstar
+    {
+    protected:
+
+        class AstarNode
+        {
+        protected:
+
+            int64_t index_;
+            int64_t backpointer_;
+            double cost_to_come_;
+            double value_;
+
+        public:
+
+            AstarNode(const int64_t index, const int64_t backpointer, const double cost_to_come, const double value) : index_(index), backpointer_(backpointer), cost_to_come_(cost_to_come), value_(value) {}
+
+            inline int64_t Index() const { return index_; }
+
+            inline int64_t Backpointer() const { return backpointer_; }
+
+            inline double CostToCome() const { return cost_to_come_; }
+
+            inline double Value() const { return value_; }
+        };
+
+        class CompareNodeFn
+        {
+            public:
+
+                constexpr bool operator()(const AstarNode& lhs, const AstarNode& rhs) const
+                {
+                    return lhs.Value() > rhs.Value();
+                }
+        };
+
+        SimpleGraphAstar() {}
+
+    public:
+
+        typedef std::pair<std::vector<int64_t>, double> AstarResult;
+
+        static AstarResult PerformAstar(const Graph<NodeValueType, Allocator>& graph, const int64_t start_index, const int64_t goal_index, const std::function<double(const NodeValueType&, const NodeValueType&)>& heuristic_fn)
```

Comment:
A high level comment describing the use of graph indices rather than state values for the AStar storage throughout would be useful I think.